OpenRA.Mods.Common/Traits/Buildings/BaseProvider.cs

```diff
@@ -45,9 +45,9 @@ public BaseProvider(Actor self, BaseProviderInfo info)
 			this.self = self;
 			devMode = self.Owner.PlayerActor.Trait<DeveloperMode>();
 			progress = total = info.InitialDelay;
-			var mapBuildRadius = self.World.WorldActor.Trait<MapBuildRadius>();
-			allyBuildEnabled = mapBuildRadius.AllyBuildRadiusEnabled;
-			buildRadiusEnabled = mapBuildRadius.BuildRadiusEnabled;
+			var mapBuildRadius = self.World.WorldActor.TraitOrDefault<MapBuildRadius>();
+			allyBuildEnabled = mapBuildRadius == null || mapBuildRadius.AllyBuildRadiusEnabled;
+			buildRadiusEnabled = mapBuildRadius == null || mapBuildRadius.BuildRadiusEnabled;
```

Comment:
IMO it would make more sense to default both of these to false if the trait isn't defined.  Likewise in `Building`.

Reply:
Makes sense. Ill change that.